### PR TITLE
[DDW-291] Smart Tooltip MVP integration

### PR DESCRIPTION
**Pull request by DominikGuzei**
This PR is the MVP for integrating the new [PopOver component](https://github.com/input-output-hk/react-polymorph/pull/150) in react-polymorph by replacing the syncing/connecting screen tooltips:

### Syncing / Connecting tooltips:

CleanShot 2020-10-21 at 14 00 35

CleanShot 2020-10-21 at 14 01 02

### Wallet Recovery Phrase Verification Tooltip:

CleanShot 2020-10-27 at 17 37 44
if there is not enough space above (in storybook):
CleanShot 2020-10-27 at 17 37 56

CleanShot 2020-10-27 at 17 50 06

### Staking Slider Tooltips:

CleanShot 2020-10-27 at 17 58 27

---

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/archives/GGKFXSKC6/p1603276278127800)

### App

- [ ] Startup the application and check that the tooltips on the syncing/connecting screen are looking and working the same as before (this PR is replacing the tooltips with a new component but the look and feel should stay the same in all themes)
- [ ] Create a Byron legacy wallet (12 words mnemonics) -> go to wallet settings and check that tooltip for "Paper wallet ?" label has the same style as before
- [ ] Go to the decentralization tab > `Stake pools` and check that tooltips for the ranking slider values (min, max, total) have acceptable styles (note: they behave differently than before because they are currently not bound by the main content section but only by the browser window edges)

### Storybook

- [ ] Go to `NODES` => `Connecting and Loading` => `Connecting` and check that the tooltips on the syncing/connecting screen are looking and working the same as before (this PR is replacing the tooltips with a new component but the look and feel should stay the same in all themes)
- [ ] Go to `WALLETS` => `Settings` => `Recovery Phrase Verification Widget` and check that tooltip for "Paper wallet ?" label has the same style as before
- [ ] Go to `DECENTRALIZATION` => `Staking` => `Pools Index` and check that tooltips for the ranking slider values (min, max, total) have acceptable styles (note: they behave differently than before because they are currently not bound by the main content section but only by the browser window edges)

### General

- [ ] Check that all tooltips have the correct background colors defined in the [design spec](https://app.zeplin.io/project/59b94291478ed0294c8db870/screen/5a0583be5abbf66a3a6b1c77)

---

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes
- [ ] PR has default-sized Daedalus window screenshots or animated GIFs of important UI changes:
  - [ ] In English
  - [ ] In Japanese
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

File: source/renderer/app/components/loading/syncing-connecting/StatusIcons.js [hunk 1]
```diff
@@ -244,29 +244,34 @@ export default class StatusIcons extends Component<Props> {
...
       <div className={styles.component}>
         {[
-          this.getIconWithToolTip(nodeStateIcon, 'nodeState'),
-          this.getIconWithToolTip(isNodeRespondingIcon, 'isNodeResponding'),
-          // this.getIconWithToolTip(isNodeSubscribedIcon, 'isNodeSubscribed'),
-          this.getIconWithToolTip(isNodeTimeCorrectIcon, 'isNodeTimeCorrect'),
-          this.getIconWithToolTip(isNodeSyncingIcon, 'isNodeSyncing'),
+          this.getIconWithPopover(nodeStateIcon, 'nodeState'),
+          this.getIconWithPopover(isNodeRespondingIcon, 'isNodeResponding'),
+          // this.getIconWithPopover(isNodeSubscribedIcon, 'isNodeSubscribed'),
```
Comment (DeeJayElly) [inline source/renderer/app/components/loading/syncing-connecting/StatusIcons.js:272]: Do we need this commented line? @DominikGuzei

Comment (DominikGuzei) [reply source/renderer/app/components/loading/syncing-connecting/StatusIcons.js:272]: It was already there before (i just updated the function name), I'm not sure if we need it - but that's out of scope of this PR

**Comment by darko-mijic** [comment]
Can we prioritize reviewing this one?

**Comment by nikolaglumac** [comment]
@daniloprates @mioriohk @gnpf please review this one today 🙏

**Comment by nikolaglumac** [comment]
@a-rukin we count on your support here! Please review ALL the tooltips 🙏

**Comment by a-rukin** [comment]
**New issues**
1 - Tooltip is cropped my left menu - see [here](https://user-images.githubusercontent.com/2539345/96993356-dff7b380-1533-11eb-9015-777802e0a57a.png)

**Comment by nikolaglumac** [comment]
@DominikGuzei please make sure not to commit these yarn.lock changes: https://github.com/input-output-hk/daedalus/pull/2210/commits/c1be0bf7f06d3f846b4420ced7ce17078197e928
...they break our CI builds and have to be _manually_ removed from yarn.lock in case they get added when you run `yarn nix:mainnet` or any other start script.

**Comment by gnpf** [comment]
@DominikGuzei For the build [14870](https://buildkite.com/input-output-hk/daedalus/builds/14870) I found no additional issues. I'll wait for the changes that @a-rukin mentioned to be included before approving.

I updated the testing summary on the description of this PR. Also included a list of the tooltips @mioriohk and I checked for this PR, please let us know if we're missing something.

**Comment by nikolaglumac** [comment]
@gnpf @mioriohk @a-rukin @ManusMcCole this PR includes only an update of the status icons on the "Connecting / Syncing" screen. Only this should be tested 🙏 

Please follow the "Testing checklist" @DominikGuzei prepared (in PR description):
Screenshot 2020-10-23 at 17 10 35

@DominikGuzei are you planning to replace other tooltips we have in the scope of this PR?

**Review by a-rukin — APPROVED** [review]

**Review by mioriohk — APPROVED** [review]

**Review by gnpf — APPROVED** [review]

**Comment by DominikGuzei** [comment]
@nikolaglumac what do you think would be smarter? (pun intended) initially, my goal was to keep the PR as simple as possible for reviewers - the potential problem with that could be that they would need to review all the tooltips for any tooltip-related changes anyway (also later on) … so my initial thought might not have been too smart in the end

**Comment by nikolaglumac** [comment]
> @nikolaglumac what do you think would be smarter? (pun intended) initially, my goal was to keep the PR as simple as possible for reviewers - the potential problem with that could be that they would need to review all the tooltips for any tooltip-related changes anyway (also later on) … so my initial thought might not have been too smart in the end

@DominikGuzei I guess we need to update and test the ones which can demonstrate how smart they are :) 
e.g. the ones which are within scrollable containers or are affected by resizing the screen, etc.

**Comment by DominikGuzei** [comment]
Good point @nikolaglumac then I include 1-2 of these to testdrive the implementation 👍

**Comment by daniloprates** [comment]
@DominikGuzei when the Paper Wallet tooltip is close to the right edge it gets cut off:

Screen Shot 2020-10-28 at 12 03 14

Screen Shot 2020-10-28 at 12 00 43

**Comment by DominikGuzei** [comment]
@daniloprates that's weird, I cannot reproduce the cut-off :( did you do anything special to get that state?

**Comment by daniloprates** [comment]
> @daniloprates that's weird, I cannot reproduce the cut-off :( did you do anything special to get that state?
> 
> grafik

The only thing I did was to force showing the Paper Wallet thing for a normal wallet (removed the `isLegacy && ...`)

**Comment by DominikGuzei** [comment]
@daniloprates ah ok, i only tested with byron legacy wallets where the Paper Wallet thing is visible 😉 can you try that too? Because your scenario would not happen "in real life" - maybe there is some other weird reason - i will try to reproduce your scenario too now

**Comment by DominikGuzei** [comment]
@a-rukin i tried to implement the pop over colors by following your design spec - however i was a bit confused because there is no naming in the design, so i had to "guess" which examples are for which theme 😉 can you double check in the latest storybook build on this PR?

**Comment by DominikGuzei** [comment]
@daniloprates I also tried your approach but I cannot reproduce the issue. Can you consistently reproduce it?

**Comment by DominikGuzei** [comment]
@a-rukin nevermind, i just saw that the [progress-loader design spec](https://app.zeplin.io/project/59b94291478ed0294c8db870/screen/5a0583be5abbf66a3a6b1c77) includes the theme names 👍

**Review by daniloprates — APPROVED** [review]
Great job 🚀

**Review by ManusMcCole — APPROVED** [review]

**Comment by nikolaglumac** [comment]
Great work @DominikGuzei 🎉

**Pull request merged**